testsuite/gna/testsuite.sh

```diff
@@ -32,27 +32,29 @@ for opt; do
   esac
 done
 
-for i in $dirs; do
-  echo "GNA dir $i:"
-  cd $i
+singlerun() {
+  echo "GNA dir $1:"
+  cd $1
   if ! ./testsuite.sh; then
     echo "#################################################################"
-    echo "######### FAILURE: $i"
+    echo "######### FAILURE: $1"
     echo "#################################################################"
-    if [ $full = "y" ]; then
-      failures="$failures $i"
+    if [ $2 = "y" ]; then
+      failures="$failures $1"
     else
       exit 1;
     fi
   fi
   cd ..
+}
+
+for i in $dirs; do
+  printf "\033[34mGNA $i \033[0m\n" 1>> ../../log.log 2>&1
+  singlerun $i $full 1>> ../../log.log 2>&1
```

Comment:
> Please, don't redirect in testsuite.sh (or add an option for that).
I often manually run then.

Working on it.

> Not sure that colours are a good idea: AFAIK they don't work on Windows. (An option is ok too).

What do you mean with 'don't work on Windows'? I have been testing them in MinGW64 (Win10), and I had no problem with them. Indeed, since I launch the same containers as in travis-ci, when I had problems with dash, colours where not shown in the Ubuntu version neither locally nor in travis-ci. But in the Fedora version, they where shown both locally and in travis-ci. Now all work.

Since I am going to define variables for colorurs, it is trivial to make them optional. However, I'd like to know if you are referring to launching those shell scripts from the PowerShell. Is it possible?

> (I understand progress is displayed on stdout and only test output is logged).

I don't get this sentence. Do you mean you understand that this is the actual behaviour, or that you expect this to be the behaviour?

If by 'progress' you mean printing a single line for each test group, that's it. If you mean printing a line for each individual test, that's not the actual behaviour. I tried it, but since everything is printed at the end, it looked redundant to me.